### PR TITLE
feat: Updating tables specification in gooddata-dbt for new API spec

### PR DESCRIPTION
**Pull request by Mara3l**
Updating the references, to be up-to-date with the current LDM specification for the API.

### CODE REVIEW

File: gooddata-dbt/gooddata_dbt/dbt/tables.py [hunk 1]
```diff
@@ -363,8 +370,13 @@ def make_references(self, table: DbtModelTable, role_playing_tables: dict) -> li
                     {
                         "identifier": {"id": referenced_object_id, "type": "dataset"},
                         "multivalue": False,
-                        "source_columns": [column.name],
-                        "source_column_data_types": [column.data_type],
+                        "sources": [
+                            {
+                                "column": column.name,
+                                "dataType": [column.data_type],
+                                "target": {"id": column.ldm_id, "type": "attribute"},
```
[the remark this answers is not in the document]
Comment (Mara3l) [reply gooddata-dbt/gooddata_dbt/dbt/tables.py]: I've updated it. Please check the updated version! :) Thank you for the idea!

File: gooddata-dbt/gooddata_dbt/dbt/tables.py [hunk 2]
```diff
@@ -344,10 +352,33 @@ def find_role_playing_tables(tables: list[DbtModelTable]) -> dict:
...
+        for table in self.tables:
+            if table.gooddata_ldm_id == referenced_object_id:
+                return self.find_primary_key(table)
+        # We must also take into consideration the role_playing_tables!
+        for name, columns in role_playing_tables.items():
+            for column in columns:
+                if f"{name}_{column}" == referenced_object_id:
+                    for table in self.tables:
+                        if table.gooddata_ldm_id == name:
+                            print("NASEL")
```
Comment (hkad98) [inline gooddata-dbt/gooddata_dbt/dbt/tables.py:371]: :)

Comment (Mara3l) [reply gooddata-dbt/gooddata_dbt/dbt/tables.py:371]: :)

File: gooddata-dbt/gooddata_dbt/dbt/tables.py [hunk 3]
```diff
@@ -358,13 +388,24 @@ def make_references(self, table: DbtModelTable, role_playing_tables: dict) -> li
                         referenced_object_id = f"{referenced_object_id}_{column.ldm_id}"
             elif column.is_date():
                 referenced_object_id = column.ldm_id
+                referenced_type = "date"
+                referenced_column_name = column.ldm_id
             if referenced_object_id is not None:
+                if referenced_column_name is None:
+                    referenced_column_name = column.meta.gooddata.referenced_column_name
+                    if referenced_column_name is None:
+                        referenced_column_name = self.find_referenced_column(referenced_object_id, role_playing_tables)
```
Comment (hkad98) [inline gooddata-dbt/gooddata_dbt/dbt/tables.py:397]:
```
referenced_column_name = column.meta.gooddata.referenced_column_name or self.find_referenced_column(referenced_object_id, role_playing_tables)
```

File: gooddata-dbt/gooddata_dbt/dbt/tables.py [hunk 4]
```diff
@@ -344,10 +352,32 @@ def find_role_playing_tables(tables: list[DbtModelTable]) -> dict:
...
+            if table.gooddata_ldm_id == referenced_object_id:
+                return self.find_primary_key(table)
+        # We must also take into consideration the role_playing_tables!
+        for name, columns in role_playing_tables.items():
+            for column in columns:
+                if f"{name}_{column}" == referenced_object_id:
+                    for table in self.tables:
+                        if table.gooddata_ldm_id == name:
+                            return f"{self.find_primary_key(table)}_{column}"
+        raise Exception
```
[the remark this answers is not in the document]
Comment (Mara3l) [reply gooddata-dbt/gooddata_dbt/dbt/tables.py:372]: Not sure if it is worth it editing it, as it might hurt the readability.

**Pull request closed without merging**